### PR TITLE
fix(design-system): eslint fix

### PR DESCRIPTION
**Pull request by frassinier**
**What is the problem this PR is trying to solve?**

More than 120+ eslint errors

**What is the chosen solution to this problem?**

Fix them all

**Please check if the PR fulfills these requirements**

- [x] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

**Review by Benrajalu — APPROVED** [review]

File: packages/design-system/src/docs/Row.tsx [hunk 1]
```diff
@@ -1,8 +1,14 @@
...
-export default function Row({ children, centered, ...props }) {
+export default function Row({
+	children,
+	centered,
+	...props
+}: React.PropsWithChildren<HTMLDivElement> & { centered: 'all' | 'vertical' | 'horizontal' }) {
 	return (
 		<div
+			/*
+			// @ts-ignore */
```
Comment (Benrajalu) [inline packages/design-system/src/docs/Row.tsx]:
```suggestion
			/* @ts-ignore */
```
But also, 😢

File: packages/design-system/src/docs/Themes.tsx [hunk 2]
```diff
@@ -132,7 +129,8 @@ const SAliasCategory = styled.article``;
 
 const previewWidth = 40;
 
-const ColorToken = ({ name, theme }) => {
+const ColorToken = ({ name, theme }: { name: string; theme: DefaultTheme }) => {
+	// @ts-ignore
```
Comment (Benrajalu) [inline packages/design-system/src/docs/Themes.tsx:133]:
```suggestion
	// @ts-ignore
```
Why? It'd be great to get into the habit of explaining the ts ignores with a comment

File: packages/design-system/src/docs/Row.tsx (hunk 1 above)
Comment (frassinier) [reply packages/design-system/src/docs/Row.tsx]: Fixed

File: packages/design-system/src/docs/Themes.tsx (hunk 2 above)
Comment (frassinier) [reply packages/design-system/src/docs/Themes.tsx]: Fixed

File: packages/design-system/src/components/Switch/Switch.tsx [hunk 3]
```diff
@@ -33,10 +33,10 @@ const Switch = ({
...
 
-	let radioWidths: number[] = [];
+	const radioWidths = useRef<number[]>();
 
 	useEffect(() => {
-		radioWidths = radio.items.map(item => {
+		radioWidths.current = radio.items.map(item => {
 			if (item.ref.current) return item.ref.current.scrollWidth;
 			return 0;
 		});
```
[the remark this answers is not in the document]
Comment (frassinier) [reply packages/design-system/src/components/Switch/Switch.tsx]: Applied

File: packages/design-system/src/docs/WithSelector.tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+		if (newClassName && newStyles.length) {
+			setClassName(newClassName);
+			setStyles(newStyles);
+		}
+	}, [ref, selector]);
+
+	React.useEffect(() => {
+		if (!ref.current) return;
+		ref.current.className = `${ref.current.className} ${className}`;
+	}, [ref, className]);
```
[the remark this answers is not in the document]
Comment (frassinier) [reply packages/design-system/src/docs/WithSelector.tsx]: ref.current is not a valid deps, I removed it

File: packages/design-system/src/docs/AlgoliaAutocomplete.tsx [hunk 5]
```diff
@@ -11,9 +11,10 @@ export function Autocomplete(props) {
 		}
 
 		const search = autocomplete({
-			container: containerRef.current,
+			container: containerRef.current || '',
 			renderer: { createElement, Fragment },
 			render({ children }, root) {
+				// @ts-ignore
```
Comment (Benrajalu) [inline packages/design-system/src/docs/AlgoliaAutocomplete.tsx:17]:
```suggestion
				// @ts-ignore
```
Missing why

File: packages/design-system/src/docs/SearchBar.tsx [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+	return (
+		<div style={{ maxWidth: '50%', margin: '0 auto' }}>
+			<Autocomplete
+				/*
+                // @ts-ignore */
+				placeholder="Search for anything on Coral"
+				getSources={
+					/*
+                    // @ts-ignore */
```
Comment (Benrajalu) [inline packages/design-system/src/docs/SearchBar.tsx:30]: Weird double comment syntax, also missing the reason why we allow these ts-ignores

File: packages/design-system/src/docs/SearchBar.tsx [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+										{
+											indexName: process.env.STORYBOOK_ALGOLIA_SEARCH_INDEX || '',
+											query,
+										},
+									],
+								});
+							},
+							templates: {
+								/*
+							// @ts-ignore */
```
Comment (Benrajalu) [inline packages/design-system/src/docs/SearchBar.tsx:47]: Same

File: packages/design-system/src/docs/Icons.tsx [hunk 8]
```diff
@@ -1,36 +1,37 @@
...
-		setTransform(event.target.value);
-	}
+	const onChangeTransform = (event: ChangeEvent<HTMLSelectElement>) => {
+		setTransform(event.currentTarget.value);
+	};
 
 	return (
 		<>
+			{/*
+			// @ts-ignore */}
```
Comment (frassinier) [inline packages/design-system/src/docs/Icons.tsx:34]:
```suggestion
			// Temporary fix, will be removed with CSS Modules conversion
			// @ts-ignore */}
```

**Review by jsomsanith-tlnd — APPROVED** [review]

**Pull request merged**